packages/backend/src/tokens/jwt/assertions.ts

```diff
@@ -43,3 +56,138 @@ export const assertAudienceClaim = (aud?: unknown, audience?: unknown) => {
+    return;
+  }
+
+  if (typ !== 'JWT') {
+    throw new TokenVerificationError({
+      action: TokenVerificationErrorAction.EnsureClerkJWT,
+      reason: TokenVerificationErrorReason.TokenInvalid,
+      message: `Invalid JWT type ${JSON.stringify(typ)}. Expected "JWT".`,
+    });
+  }
+};
+
+export const assertHeaderAlgorithm = (alg: string) => {
+  if (!algToHash[alg]) {
+    throw new TokenVerificationError({
+      action: TokenVerificationErrorAction.EnsureClerkJWT,
+      reason: TokenVerificationErrorReason.TokenInvalidAlgorithm,
+      message: `Invalid JWT algorithm ${JSON.stringify(alg)}. Supported: ${algs}.`,
+    });
+  }
+};
+
+export const assertSubClaim = (sub?: string) => {
+  if (typeof sub !== 'string') {
+    throw new TokenVerificationError({
+      action: TokenVerificationErrorAction.EnsureClerkJWT,
+      reason: TokenVerificationErrorReason.TokenVerificationFailed,
+      message: `Subject claim (sub) is required and must be a string. Received ${JSON.stringify(sub)}.`,
+    });
+  }
+};
+
+export const assertAuthorizedPartiesClaim = (azp?: string, authorizedParties?: string[]) => {
+  if (!azp || !authorizedParties || authorizedParties.length === 0) {
+    return;
+  }
+
+  if (!authorizedParties.includes(azp)) {
+    throw new TokenVerificationError({
+      reason: TokenVerificationErrorReason.TokenInvalidAuthorizedParties,
+      message: `Invalid JWT Authorized party claim (azp) ${JSON.stringify(azp)}. Expected "${authorizedParties}".`,
+    });
+  }
+};
+
+export const assertIssuerClaim = (iss: string, issuer: IssuerResolver | null) => {
+  if (typeof issuer === 'function' && !issuer(iss)) {
+    throw new TokenVerificationError({
+      reason: TokenVerificationErrorReason.TokenInvalidIssuer,
+      message: 'Failed JWT issuer resolver. Make sure that the resolver returns a truthy value.',
+    });
+  } else if (typeof issuer === 'string' && iss && iss !== issuer) {
+    throw new TokenVerificationError({
+      reason: TokenVerificationErrorReason.TokenInvalidIssuer,
+      message: `Invalid JWT issuer claim (iss) ${JSON.stringify(iss)}. Expected "${issuer}".`,
+    });
+  }
+};
+
+export const assertExpirationClaim = (exp: number, clockSkewInMs: number) => {
+  if (typeof exp !== 'number') {
+    throw new TokenVerificationError({
+      action: TokenVerificationErrorAction.EnsureClerkJWT,
+      reason: TokenVerificationErrorReason.TokenVerificationFailed,
+      message: `Invalid JWT expiry date claim (exp) ${JSON.stringify(exp)}. Expected number.`,
+    });
+  }
+
+  const currentDate = new Date(Date.now());
+  const expiryDate = new Date(0);
+  expiryDate.setUTCSeconds(exp);
+
+  const expired = expiryDate.getTime() <= currentDate.getTime() - clockSkewInMs;
+  if (expired) {
+    throw new TokenVerificationError({
+      reason: TokenVerificationErrorReason.TokenExpired,
+      message: `JWT is expired. Expiry date: ${expiryDate.toUTCString()}, Current date: ${currentDate.toUTCString()}.`,
```

Comment:
☁️ i've converted the claims and the current dates to UTC in the error messages to improve the readability of the error message. If you think that by leaving the current date to be printed with the timezone information of the machine that is being executed will help us identify any issue, let me know and I will revert this change.

Reply:
@dimkl I agree with this change, good call!